exercises/sgf-parsing/description.md

```diff
@@ -53,6 +53,18 @@ A key can have multiple values associated with it. For example:
 
 Here `AB` (add black) is used to add three black stones to the board.
 
+According the the SGF specification `\` is the escape character. When `\` is
+encountered during parsing, it indicates that the following character should be
+printed as is. The exception to this is whitespace, with newline, carriage
+return and tab encoded as `"\n"`, `"\r"` and `"\t"` respectively. When parsing
+you can expect some characters to _always_ be escaped, including parentheses,
+square brackets, semicolons and colons.
```

Comment:
I'm unsure about the sentence beginning with "When parsing you can expect..." for two reasons:

1. https://www.red-bean.com/sgf/sgf4.html does not say that parentheses or semicolons will always be escaped, so if we make this claim, we are going against the specification... but that is okay! It's okay for this exercise's description to be **more restrictive** than the specification. It just means that we'll never test some inputs that would be legal, and that's fine. If we were **less restrictive** than the specification, then we would test inputs that would be illegal, and we should never do that. So if we're keeping this sentence as-is, let us just agree here that we have decided to deliberately be more restrictive than the specification, to make things easier on the students. Please confirm.
2. It's not clear to me that the student's job is made easier by knowing this. I guess it means that the student can immediately search for the next non-text character they expect to see (that would typically be `]` in the inputs we're testing), and if they find it's been escaped, they can just search for the next one, repeating until they find a non-escaped one? Sure I guess, that's fine. I suppose it doesn't hurt.

Reply:
1. yes the specification didn't say that semicolon or parenthesis must be escaped, but I don't understand how you can practically include those chars in the text without escaping them (you may be able to so closing parens without escaping but not semicolon or open parens) so I assumed its a typo. the spec does say that colons must be escaped which I suspect is a type intending semicolon but I could be wrong there so I included colons.
2. it seems to me the student needs to know how to handle these chars when encountered in text. if in theory these chars could appear unescaped it make the task much more complex so I think this information is very helpful

Reply:
Huh, sounds like they would be interesting test cases to add, then... This statement is **not** a request that the test case should be added as a part of this PR, it was just a statement in passing of what might make a good future addition.

The reason you can have an unescaped semicolon inside a Text is that Texts are only closed by the specific character `]`, so if you've encountered a `;` before a `]` you know the semicolon is part of the Text and not the semicolon that indicates a new node. Same reasoning for the other characters.

Reply:
i see your point